compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -544,6 +544,12 @@ object Checking {
       else if !sym.owner.isClass && !(sym.is(Param) && sym.owner.isConstructor) then fail(em"@publicInBinary cannot be used on local definitions")
       else if sym.is(ParamAccessor) && sym.is(Private) then fail(em"@publicInBinary cannot be non `val` constructor parameters")
       else if sym.is(Private) && !sym.privateWithin.exists && !sym.isConstructor then fail(em"@publicInBinary cannot be used on private definitions\n\nConsider using `private[${sym.owner.name}]` or `protected` instead")
+    else
+      sym.allOverriddenSymbols.find(sym => sym.hasAnnotation(defn.PublicInBinaryAnnot)) match
```

Comment:
As mentioned by @smarter somewhere else, there's a good chance that this code would behave much better in `RefChecks`. One of the main purposes of `RefChecks` is to check properties across the inheritance chain of methods, so it would make sense there.

Reply:
I made the change. I will test the community build with the CI.

Reply:
It worked